src/services.js

```diff
@@ -587,6 +587,17 @@ export class Services {
       getServiceForDoc(elementOrAmpDoc, 'video-manager')));
   }
 
+  /**
+   * @param {!Window} win
+   * @return {!Promise<?../extensions/amp-viewer-assistance/0.1/amp-viewer-assistance.AmpViewerAssistance>}
+   */
+  static viewerAssistanceForDocOrNull(win) {
+    return (
+    /** @type {!Promise<?../extensions/amp-viewer-assistance/0.1/amp-viewer-assistance.AmpViewerAssistance>} */
+      (getElementServiceIfAvailable(win, 'amp-viewer-assistance',
+          'amp-viewer-assistance', true)));
```

Comment:
Does this work? Looks like amp-viewer-assistance.js calls `registerServiceForDoc`, which means we should be using `getElementServiceIfAvailableForDoc`.

Reply:
It works yes, but maybe you can help me understand the difference between the two functions.

Also, there is not an element for `ampViewerAssistance` but instead a script as discussed [here](https://github.com/ampproject/amphtml/pull/20725#discussion_r259143957). What would be passed in instead?

Reply:
Hm, curious why this works. You can simply pass in the `element` in `getViewerAuthTokenIfAvailable`.

The `element` param is just used for service scoping and doesn't need to belong to the extension. In some cases, an AMP document can be embedded inside a non-AMP page via shadow DOM (which shares the same `window`), so we need to pass a local element instead. 

Check out the usages of `bindForDocOrNull` as an example. There's no such thing as an `<amp-bind>` element. Rather, the Bind service is scoped to the "doc" level rather than "window" level.

Reply:
That makes sense. Thanks for the explanation.